src/utils/solvePuzzle.ts

```diff
@@ -0,0 +1,131 @@
+import update from 'immutability-helper';
+import { MAX_PRONGS } from '../constants';
+import { DigiKey } from '../types/DigiKey';
+import { Puzzle } from '../types/Puzzle';
+import { rotateKey } from './rotateKey';
+
+class Key implements DigiKey {
```

Comment:
Can we avoid classes? It looks like we can replace this with a plain old interface.

Classes are only useful if we need to have specific constructor logic.

Reply:
ah yeah remnants of a more complex solution i was working on :P it had like 6 body methods, will do!